src/charm.py

```diff
@@ -377,6 +411,28 @@ def _construct_waypoint(self):
             spec=gateway.spec.model_dump(),
         )
 
+    def _construct_hpa(self, unit_count: int) -> HorizontalPodAutoscaler:
+        """Constructs a HorizontalPodAutoscaler resource targeting the waypoint Deployment.
+
+        This HPA is used to scale the waypoint workload automatically when the charm scales.
+        The scaling is achieved by setting the min and max replica settings in the HPA to be the same as the unit count. It is important to note that the HPA must target the waypoint's Deployment and not Gateway.
```

Comment:
something isn't right with the linting on this repo, but it should have complained about line width here...

```suggestion
        The scaling is achieved by setting the min and max replica settings in the HPA to be the same as the unit count.
        It is important to note that the HPA must target the waypoint's Deployment and not Gateway.
```